### DISCUSSION TITLE
Oauth2 : Add the oauth2 client credentials bearer

### DISCUSSION THREAD

**Discussion by sylvainmouquet** [fastapi/fastapi#8570] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
oauth2_clientcredentials_scheme = OAuth2ClientCredentialsBearer(
    tokenUrl="oauth/clientcredentials_token",
    description="My client_id and my client_credentials",
    auto_error=False
)
```

### Description

In Swagger UI, there are 4 types of oauth2 flow. 
The client credentials flow is not present in FastAPI

https://swagger.io/docs/specification/authentication/oauth2/

### Wanted Solution

Add client credentials flow in oauth2.py

### Wanted Code

```python
oauth2_clientcredentials_scheme = OAuth2ClientCredentialsBearer(
    tokenUrl="oauth/clientcredentials_token",
    description="My client_id and my client_credentials",
    auto_error=False
)
```

### Operating System

macOS

### FastAPI Version

0.73.0

### Python Version

3.8.7

**Comment by sylvainmouquet** [comment]
Any news?

**Comment by waldner** [comment]
Any reason why this is not included?

**Comment by milkshakeuk** [comment]
@sylvainmouquet looks like this is incoming soon #11560
